src/harness/harness.ts

```diff
@@ -1031,146 +969,132 @@ namespace Harness {
+            currentDirectory: string): CompilationOutput {
 
-                // Files from built\local that are requested by test "@includeBuiltFiles" to be in the context.
-                // Treat them as library files, so include them in build, but not in baselines.
-                const includeBuiltFiles: TestFileWithPath[] = [];
-                if (options.includeBuiltFile) {
-                    const builtFileName = libFolder + options.includeBuiltFile;
-                    const builtFile: TestFileWithPath = {
-                        unitName: builtFileName,
-                        content: normalizeLineEndings(IO.readFile(builtFileName), newLine),
-                        path: ts.toPath(builtFileName, currentDirectory, getCanonicalFileName)
-                    };
-                    includeBuiltFiles.push(builtFile);
-                }
+            const options: ts.CompilerOptions & HarnessOptions = compilerOptions ? ts.clone(compilerOptions) : { noResolve: false };
+            options.target = options.target || ts.ScriptTarget.ES3;
+            options.module = options.module || ts.ModuleKind.None;
+            options.newLine = options.newLine || ts.NewLineKind.CarriageReturnLineFeed;
+            options.noErrorTruncation = true;
+            options.skipDefaultLibCheck = true;
 
-                const fileOutputs: GeneratedFile[] = [];
+            const newLine = "\r\n";
+            currentDirectory = currentDirectory || Harness.IO.getCurrentDirectory();
 
-                const programFiles = inputFiles.concat(includeBuiltFiles).map(file => file.unitName);
+            // Parse settings
+            let useCaseSensitiveFileNames = Harness.IO.useCaseSensitiveFileNames();
+            if (harnessSettings) {
+                setCompilerOptionsFromHarnessSetting(harnessSettings, options);
+            }
+            if (options.useCaseSensitiveFileNames !== undefined) {
+                useCaseSensitiveFileNames = options.useCaseSensitiveFileNames;
+            }
 
-                const compilerHost = createCompilerHost(
-                    inputFilesWithPath.concat(includeBuiltFiles).concat(otherFilesWithPath),
-                    (fn, contents, writeByteOrderMark) => fileOutputs.push({ fileName: fn, code: contents, writeByteOrderMark: writeByteOrderMark }),
-                    options.target, useCaseSensitiveFileNames, currentDirectory, options.newLine);
-                const program = ts.createProgram(programFiles, options, compilerHost);
+            // Files from built\local that are requested by test "@includeBuiltFiles" to be in the context.
+            // Treat them as library files, so include them in build, but not in baselines.
+            const includeBuiltFiles: TestFile[] = [];
+            if (options.includeBuiltFile) {
+                const builtFileName = libFolder + options.includeBuiltFile;

```

Comment:
If `path` is present in this file, use [`path.join`](https://nodejs.org/api/path.html#path_path_join_path1_path2)